lib/unleash/strategies.rb

```diff
@@ -64,5 +70,11 @@ def register_base_strategies
       Unleash::Strategy::RemoteAddress,
       Unleash::Strategy::UserWithId
     ].freeze
+
+    def warn_deprecated_registration(strategy, method)
+      warn "[DEPRECATED] Registering custom Unleash strategy by #{method} is deprecated.
```

Comment:
We currently have a logger instance where all output goes to. (in case someone would like to suppress warnings, for whatever reason). Would be nice to keep using it.

Reply:
For deprecation warnings, it is common to use the Kerner warning mechanism as it allows engineers to see warnings regardless of local logger configuration. Example:
- [Gem native way to notify deprecation](https://docs.ruby-lang.org/en/2.1.0/Gem/Deprecate.html)
- [Rails default deprecation behavior](https://apidock.com/rails/ActiveSupport/Deprecation/Behavior/behavior)

In this case, I decided to simply use `Kernel.warn` as Gem related functionality does not allow creating a custom message for the warning.